zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/util/OpensearchRepository.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.tasks.util;
+
+import io.camunda.zeebe.exporter.api.ExporterException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.opensearch.client.opensearch.OpenSearchAsyncClient;
+import org.opensearch.client.opensearch._types.ErrorCause;
+import org.opensearch.client.opensearch._types.Time;
+import org.opensearch.client.opensearch.core.ClearScrollRequest;
+import org.opensearch.client.opensearch.core.SearchRequest;
+import org.opensearch.client.opensearch.core.bulk.BulkResponseItem;
+import org.opensearch.client.opensearch.core.search.Hit;
+import org.slf4j.Logger;
+
+public class OpensearchRepository {
+
+  private final Time SCROLL_KEEP_ALIVE = Time.of(t -> t.time("1m"));
+  private final int SCROLL_PAGE_SIZE = 100;
+
+  /**
+   * Variant of {@link #fetchUnboundedDocumentCollection(OpenSearchAsyncClient, Executor, Logger,
+   * SearchRequest.Builder, Class, Function)} to use when you don't care about the source document,
+   * meaning you won't be using any deserialization functionality.
+   */
+  public <T> CompletionStage<Collection<T>> fetchUnboundedDocumentCollection(
+      final OpenSearchAsyncClient client,
+      final Executor executor,
+      final Logger logger,
+      final SearchRequest.Builder requestBuilder,
+      final Function<Hit<Object>, T> transformer) {
+    return fetchUnboundedDocumentCollection(
+        client, executor, logger, requestBuilder, Object.class, transformer);
+  }
+
+  public <TDocument, TResult> CompletionStage<Collection<TResult>> fetchUnboundedDocumentCollection(
+      final OpenSearchAsyncClient client,
```

Comment:
:wrench: given the class is called "*Repository" I would pass the dependencies (OpenSearchAsyncClient, Executor, Logger)  in the constructor